src/plugin.js

```diff
@@ -300,18 +300,16 @@ const contribAdsPlugin = function(options) {
     // * An asynchronous ad request is ongoing while content is playing
     // * A non-linear ad is active
     isInAdMode() {
-
-             // Saw "play" but not "adsready"
-      return player.ads.state === 'ads-ready?' ||
-
-             // Waiting to learn about preroll
-             player.ads.state === 'preroll?' ||
-
-             // A linear ad is active
-             player.ads.state === 'ad-playback' ||
-
-             // Content is not playing again yet
-             player.ads.state === 'content-resuming';
+      if (player.ads.state === 'ads-ready?' || // Waiting to learn about preroll
+          player.ads.state === 'preroll?' || // A linear ad is active
+          player.ads.state === 'ad-playback' || // Content is not playing again yet
+          player.ads.state === 'content-resuming') {
+        player.addClass('vjs-waiting');
+        return true;
```

Comment:
We should instead store the value of the boolean expression and check it's value for whether to add or remove the class and then return it rather than return booleans directly.